product/specs/2022-01-views/04-ui-requirements-for-views.md

```diff
@@ -0,0 +1,78 @@
+---
+title: Product Specs
+description: Views: UI Requirements for Views
+published: true
+date: 2022-01-13T19:49:54Z
+tags: 
+editor: markdown
+dateCreated: 2022-01-13T19:49:54Z
+---
+
+Based on the models defined for [views](/product/specs/2022-01-views/02-modeling-views.md) and their [columns](/product/specs/2022-01-views/03-modeling-view-columns.md), the first version of Views should support the following features.
+
+I've marked the features we should support as "for alpha release", but I've also listed a few potential future features so that we can think about them while designing code architecture and UI/UX.
+
+## Query
+- **For alpha release**: Users should be able to see the query associated with a View after it has been loaded.
+- **Potential future features**:
+	- Users can edit the query associated with a View.
+	- Users can create a new View using a query.
+
+## Columns
+- **For alpha release**: Users should be able to see all columns associated with a view. Each column should show:
+	- Data Type (non-editable)
+	- Data Sources (editable if there's no Formula)
+	- Data Formula (editable in simple cases)
+- **Potential future features**:
+	- Show virtual columns involved in view creation (via CTEs, subqueries, etc.)
+	- Allow editing data formula using SQL
+
+## Rows
+- **For alpha release**: 
+	- Users should be able to see the rows associated with a view.
+	- If a cell is a direct representation of a record, users should be able to edit that record via that cell.
+	- If a column is a direct representation of a record, users should be able to add a new record via that cell.
```

Comment:
To double-check; the intent would be to pop up some "record input" UI when trying to edit the cell, correct?  I.e,. since the entire table wouldn't be there, they'd need some interface to input the rest of the row in the associated table.  What if the column is the join column?  Would they edit the record in both tables, or just one?  (Or does that not count as a direct representation?)

Reply:
Yes, that's the idea.

I think if it's a join column, it should be updated in both tables. I'll specify.

Reply:
Actually, on second thought, join columns should not count as direct representations. Only columns with a single data source count.